presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetHiveRecordCursor.java

```diff
@@ -771,15 +876,26 @@ public void start()
                 if (nullBuilder == null || (nullBuilder.getPositionCount() >= NULL_BUILDER_POSITIONS_THRESHOLD && nullBuilder.getSizeInBytes() >= NULL_BUILDER_SIZE_IN_BYTES_THRESHOLD)) {
                     nullBuilder = rowType.createBlockBuilder(new BlockBuilderStatus(), NULL_BUILDER_POSITIONS_THRESHOLD);
                 }
-                currentEntryBuilder = nullBuilder.beginBlockEntry();
+                currentEntryBuilder = nullBuilder;
             }
             else {
                 while (builder.getPositionCount() < fieldIndex) {
                     builder.appendNull();
                 }
-                currentEntryBuilder = builder.beginBlockEntry();
+                currentEntryBuilder = builder;
             }
+
+            if (tmpBuilder != null) {
```

Comment:
previous 2 changes could avoid, just update currentEntryBuilder if tmpBuilder is not null